DLT Workshop

# Question 1: What's the version of dlt
Answer: 1.14.1

In [1]:
!dlt --version

dlt 1.14.1


In [4]:
import requests
import dlt
from dlt.destinations import qdrant

In [5]:
@dlt.resource
def zoomcamp_data():
    docs_url = 'https://github.com/alexeygrigorev/llm-rag-workshop/raw/main/notebooks/documents.json'
    docs_response = requests.get(docs_url)
    documents_raw = docs_response.json()

    for course in documents_raw:
        course_name = course['course']

        for doc in course['documents']:
            doc['course'] = course_name
            yield doc

In [6]:
qdrant_destination = qdrant(
  qd_path="db.qdrant",
)

In [7]:
pipeline = dlt.pipeline(
    pipeline_name="zoomcamp_pipeline",
    destination=qdrant_destination,
    dataset_name="zoomcamp_tagged_data"

)
load_info = pipeline.run(zoomcamp_data())
print(pipeline.last_trace)

Run started at 2025-07-17 01:30:39.097244+00:00 and COMPLETED in 10.24 seconds with 4 steps.
Step extract COMPLETED in 1.40 seconds.

Load package 1752715842.287961 is EXTRACTED and NOT YET LOADED to the destination and contains no failed jobs

Step normalize COMPLETED in 0.17 seconds.
Normalized data for the following tables:
- _dlt_pipeline_state: 1 row(s)
- zoomcamp_data: 948 row(s)

Load package 1752715842.287961 is NORMALIZED and NOT YET LOADED to the destination and contains no failed jobs

Step load COMPLETED in 5.58 seconds.
Pipeline zoomcamp_pipeline load step completed in 5.56 seconds
1 load package(s) were loaded to destination qdrant and into dataset zoomcamp_tagged_data
The qdrant destination used /workspaces/llm-zoomcamp/jg_homework/dlt_workshop/db.qdrant location to store data
Load package 1752715842.287961 is LOADED and contains no failed jobs

Step run COMPLETED in 10.24 seconds.
Pipeline zoomcamp_pipeline load step completed in 5.56 seconds
1 load package(s) were load

# Question 2: How many rows were inserted into the `zoomcamp_data` collection?
# Answer: 948 row(s)

# Question 3: 
# Answer: fast-bge-small-en